src/backend/src/services/teams.services.ts

```diff
@@ -272,8 +288,12 @@ export default class TeamsService {
       throw new HttpException(400, 'A lead cannot be the head of the team!');
     }
 
+    // removes the new leads as current members of the given team (if they are current members of that team)
     if (team.members.map((member) => member.userId).some((memberId) => userIds.includes(memberId))) {
-      throw new HttpException(400, 'A lead cannot be a member of the team!');
+      const newMembersArr: number[] = team.members
+        .map((member) => member.userId)
+        .filter((memberId) => !userIds.includes(memberId));
```

Comment:
you should make a `removeUserFromTeam` helper method for this map and filter you're calling each time